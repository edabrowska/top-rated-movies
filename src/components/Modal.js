import React, { useRef } from 'react'
import { connect } from 'react-redux'
import bemCx from 'bem-modifiers'

import { useOnClickOutside } from '../hooks/useOnClickOutside'

import { closeMovieModal } from '../store/actions/index'

import ProductionCompany from './ProductionCompany'

const ConnectedModal = ({ movie, closeMovieModal }) => {

  if (Object.entries(movie).length === 0) return null

  const {
    id,
    poster_path,
    title,
    release_date,
    vote_average,
    overview,
    genres,
    production_companies
  } = movie

  const modalContent = useRef(null)
  useOnClickOutside(modalContent, closeMovieModal)

  console.log(movie)

  return (
    <div className={bemCx('modal__page-overlay', id
      ? 'overlay-in'
      : 'overlay-out'
    )}>
      <div
        className={bemCx('modal', id ? 'modal-in' : 'modal-out')}
        ref={modalContent}
      >
        <div className='modal__img-wrapper'>
          <img
            src={`https://image.tmdb.org/t/p/w500/${poster_path}`}
            alt={`${title} poster image`}
            className='modal__poster-image'
          />
        </div>
        <div className='modal__content'>
          <div className='modal__header'>
            <div className='modal__title-wrapper'>
              <h3 className='modal__title'>{title}</h3>
              <p className='modal__release-date'>{release_date}</p>
            </div>
            <p className='modal__vote'>{vote_average}</p>
          </div>
          <div className='modal__overview'>{overview}</div>
          <div className='modal__genre'>
            <p className='modal__genre-title'>Genres:</p>
            <div className='modal__genre-wrapper'>
              {genres.map(genre => <p key={genre.id} className='modal__genre-type'>{genre.name}</p>)}
            </div>
          </div>
          <div className='modal__production-company'>
            {production_companies.map(company => <ProductionCompany
              key={company.id}
              logo={company.logo_path}
              name={company.name}
            />)}
          </div>
        </div>
        <button className='modal__button' onClick={closeMovieModal}>asdXXX</button>
      </div>
    </div>
  )
}

const Modal = connect(
  state => ({
    movie: state.movieDetails.movie || {}
  }),
  dispatch => ({
    closeMovieModal: () => dispatch(closeMovieModal())
  })
)(ConnectedModal)

export default Modal
