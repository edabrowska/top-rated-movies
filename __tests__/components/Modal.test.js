import React from 'react'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'

import initialState from '../../__mocks__/fileMock'

import Modal from '../../src/components/Modal'

describe('Modal', () => {
  const movie = initialState.movieDetails.movie
  const mockStore = configureStore()

  it('Snapshot', () => {
    const component = renderer.create(
      <Provider store={mockStore(initialState)}>
        <Modal movie={movie} />
      </Provider>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
