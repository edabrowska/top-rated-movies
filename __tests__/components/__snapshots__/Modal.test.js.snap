// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Snapshot 1`] = `
<div
  className="modal__page-overlay modal__page-overlay--overlay-in"
>
  <div
    className="modal modal--modal-in"
  >
    <div
      className="modal__img-wrapper"
    >
      <img
        alt="Dolittle poster image"
        className="modal__poster-image"
        src="https://image.tmdb.org/t/p/w500//3Nt3v1uzUgfSuVARD1AnI9g9Zl9.jpg"
      />
    </div>
    <div
      className="modal__content"
    >
      <div
        className="modal__header"
      >
        <div
          className="modal__title-wrapper"
        >
          <h3
            className="modal__title"
          >
            Dolittle
          </h3>
          <p
            className="modal__release-date"
          >
            2020-01-01
          </p>
        </div>
        <p
          className="modal__vote"
        >
          6.4
        </p>
      </div>
      <div
        className="modal__overview"
      >
        After losing his wife seven years earlier, the eccentric Dr. John Dolittle, famed doctor and veterinarian of Queen Victoriaâ€™s England, hermits himself away behind the high walls of Dolittle Manor with only his menagerie of exotic animals for company. But when the young queen falls gravely ill, a reluctant Dolittle is forced to set sail on an epic adventure to a mythical island in search of a cure, regaining his wit and courage as he crosses old adversaries and discovers wondrous creatures.
      </div>
      <div
        className="modal__genre"
      >
        <p
          className="modal__genre-title"
        >
          Genres:
        </p>
        <div
          className="modal__genre-wrapper"
        >
          <p
            className="modal__genre-type"
          >
            Family
          </p>
          <p
            className="modal__genre-type"
          >
            Comedy
          </p>
        </div>
      </div>
      <div
        className="modal__production-company"
      >
        <div
          className="production-company production-company--without-logo"
        >
          <p
            className="production-company__name"
          >
            Universal Pictures
          </p>
        </div>
        <div
          className="production-company production-company--without-logo"
        >
          <p
            className="production-company__name"
          >
            Team Downey
          </p>
        </div>
        <div
          className="production-company production-company--without-logo"
        >
          <p
            className="production-company__name"
          >
            Perfect World Pictures
          </p>
        </div>
      </div>
    </div>
    <button
      className="modal__button"
      onClick={[Function]}
    >
      asdXXX
    </button>
  </div>
</div>
`;
